idea/src/org/jetbrains/kotlin/idea/intentions/ConvertToBlockBodyIntention.kt

```diff
@@ -56,10 +60,24 @@ class ConvertToBlockBodyIntention : SelfTargetingIntention<KtDeclarationWithBody
                 val factory = KtPsiFactory(declaration)
                 if (bodyType != null && bodyType.isUnit() && body is KtNameReferenceExpression) return factory.createEmptyBody()
                 val unitWhenAsResult = (bodyType == null || bodyType.isUnit()) && body.resultingWhens().isNotEmpty()
-                val needReturn = returnsValue &&
-                        (bodyType == null || (!bodyType.isUnit() && !bodyType.isNothing()))
-                val statement = if (needReturn || unitWhenAsResult) factory.createExpressionByPattern("return $0", body) else body
-                return factory.createSingleStatementBlock(statement)
+                val needReturn = returnsValue && (bodyType == null || (!bodyType.isUnit() && !bodyType.isNothing()))
+                return if (needReturn || unitWhenAsResult) {
+                    val annotatedExpr = body as? KtAnnotatedExpression
+                    val suppressAnnotation = annotatedExpr?.annotationEntries?.singleOrNull()?.takeIf {
+                        val context = annotatedExpr.analyze(BodyResolveMode.PARTIAL)
+                        context[BindingContext.ANNOTATION, it]?.type?.fqName == FqName("kotlin.Suppress")
```

Comment:
The problem happens not only for `suppress` annotation but for other ones too, so there is no need to filter them out